src/csm_testing/tests/sat_functional/bootprep/test_bootprep.py

```diff
@@ -723,14 +740,91 @@ def test_configs_images_and_session_templates(self):
+        self.assertEqual(1, len(overwrite_report['session_templates']))
+
+        self.cleanup_items(overwrite_report)
+
+    def test_cfs_version(self):
+        """Test running bootprep with cfs v2"""
+        # Not adding test for cfs v3 since that is used by default
+        result = self.run_bootprep('configs-images-and-session-templates.yaml', '--format json --cfs-version v2')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual(1, len(report['configurations']))
+        self.assertEqual(1, len(report['images']))
+        self.assertEqual(1, len(report['session_templates']))
+
         for config in report['configurations']:
-            self.items_to_delete['configurations'].append(config['name'])
+            self.validate_cfs_config(config['name'])
 
-        for image in report['images']:
-            self.items_to_delete['images'].append(image['final_image_id'])
+        self.cleanup_items(report)
+
+    def test_dry_run_and_save(self):
+        """Test running bootprep in dry-run and saving files"""
+        bootprep_opts = '--format json '
+        bootprep_opts += '--dry-run '
+        bootprep_opts += '--save-files '
+        bootprep_opts += f'-o .'
+        result = self.run_bootprep('configs-images-and-session-templates.yaml', bootprep_opts)
+
+        self.assert_in_log_messages(
+            "INFO",
+            "Would create 1 CFS configuration",
+            result.stderr.decode()
+        )
+        self.assert_in_log_messages(
+            "INFO",
+            "Would create 1 images.",
+            result.stderr.decode()
+        )
+        self.assert_in_log_messages(
+            "INFO",
+            "Would create 1 BOS session template",
+            result.stderr.decode()
+        )
+
+        cfs_config_file = f'cfs-configuration-{self.test_prefix}-simple-passing-configuration.json'
+        session_template_file = f'bos-session-template-{self.test_prefix}-simple-session-template.json'
+
+        self.assertTrue(os.path.isfile(os.path.join(self.temp_dir.name, cfs_config_file)))
+        self.assertTrue(os.path.isfile(os.path.join(self.temp_dir.name, session_template_file)))
+
+    def test_limit_option(self):
+        """Test limit option when creating items"""
+        bootprep_options = '--format json --limit configurations'
+        limit_configs_result = self.run_bootprep('configs-images-and-session-templates.yaml', bootprep_options)
+        bootprep_options = '--format json --limit images'
+        limit_images_result = self.run_bootprep('configs-images-and-session-templates.yaml', bootprep_options)
+        bootprep_options = '--format json --limit session_templates'
+        limit_session_templates_result = self.run_bootprep('configs-images-and-session-templates-overwrite.yaml', bootprep_options)
+
+        configs_report = json.loads(limit_configs_result.stdout.decode())
+        self.assertEqual(1, len(configs_report['configurations']))
```

Comment:
For each of these results, it would be good to also check that the other values are empty. That is, for this one, check that `configs_report['images']` and `configs_report['session_templates']` are empty.